src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -372,6 +376,10 @@ void DoBindType(NpgsqlTypeMapping mapping, NpgsqlConnector connector)
                 ? DatabaseInfo.ByName.TryGetValue(pgName, out var pgType)  // No dot, partial type name
                 : DatabaseInfo.ByFullName.TryGetValue(pgName, out pgType); // Full type name with namespace
 
+            // CrateDB does not support composite types.
+            if (!found && connector.IsCrateDB)
```

Comment:
Since you've already extended DatabaseInfo to have a CrateDB-specific version, you may as well override `DatabaseInfo.TryGetComposite()` to not do anything - this would remove the CrateDB check here.